### PR TITLE
#159204715 Users should get JWT on successful registration

### PR DESCRIPTION
**Pull request by Veraclins**
#### What does this PR do?
Add Authentication through JWTs and a Middleware to protect auth-only routes
#### Description of Task to be completed?
- add support for signing and verifying JWTs
- add a middleware to protect auth-only routes
- add ES6 support for tests and coverage using babel-polyfill
- add mocha support to eslintrc.js (prevents Eslint from flagging describe() and it() as undefined)
- modify .travis.yml to enable tests to run on travis
#### How should this be manually tested?
- Clone the repo: Run `git clone https://github.com/andela/elven-ah.git` in the folder where you want the repo to be saved.
- Navigate into the newly created folder and install the dependencies: `cd elven-ah && npm install`
- Copy .env.example to .env: `cp .env.example .env`
- Provision a PostgreSQL database and add the credentials to the .env file. 
- Start the server using `npm start`
- Make a POST request to http://localhost:3000/api/users/testUser using postman submitting the following fields in the body of x-www-form-urlencoded or as raw JSON:
    - username
    - email
    - password
- add the field `x-access-token` (with the token received above as value) to request header
- send a GET request to http://localhost:3000/api/users
#### What are the relevant pivotal tracker stories? 
[#159204715](https://www.pivotaltracker.com/story/show/159204715)
#### Screenshots
screen shot 2018-07-29 at 5 16 52 pm

### CODE REVIEW

**Review by madeofhuman — CHANGES_REQUESTED** [review]
You did a nice work, Clinton! I especially love how you commented your methods with jsdoc. Keep it up as it'll help other devs seamlessly onboard onto your code.

However, I just noticed that your controller naming is in the plural form which does not really follow convention (you have `UsersController.js` instead of `UserController.js`, and `ArticlesController.js` instead of `ArticleController.js`. You can push a commit that fixes that into this PR, and try as much as you can to follow the agreed-upon conventions, as they'll make our code look seamless.

File: .travis.yml [hunk 1]
```diff
@@ -8,15 +8,16 @@ services:
 addons:
   chrome: stable
 before_script:
-  - npm install nyc
+  - npm install -g nyc sequelize-cli
```
Comment (tonymontaro) [inline .travis.yml]: Any reason why 'nyc' and 'sequelize-cli' are being installed globally.

File: server/controllers/ArticleController.js [hunk 2]
```diff
@@ -1,18 +1,18 @@
 /**
   * This class contains all the methods responsible for creating and querying
-  * user profiles on the app
+  * articles on the app
   * It is made up static methods which can be called from anywhere in the app.
   */
 export default class ArticlesController {
```
Comment (madeofhuman) [inline server/controllers/ArticleController.js]: I noticed you changed the name of the file. Well done. However, your class name is still pluralised. Please review and make the necessary change as soon as you can.

File: server/controllers/UserController.js [hunk 3]
```diff
@@ -54,12 +56,23 @@ export default class UsersController {
...
+    User.findOne({
+      where: { username },
+      attributes: ['id', 'username', 'email', 'bio', 'image'],
+    })
+      .then(user => res.status(200).send({
+        status: 'success',
+        user,
+      }));
     // to be implemented
   }
```
Comment (madeofhuman) [inline server/controllers/UserController.js:78]: I noticed you changed the name of the file. Well done. However, your class name is still pluralised. Please review and make the necessary change as soon as you can.

File: server/routes/users.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import { Router } from 'express';
-import UsersController from '../controllers/UsersController';
+import UsersController from '../controllers/UserController';
```
Comment (madeofhuman) [inline server/routes/users.js]: You also need to change your imports accordingly after renaming the class, to avoid pointing to an unexisting object.

**Review by madeofhuman — CHANGES_REQUESTED** [review]
@Veraclins you've really done a good job so far. Well done. I've left some reviews for you to go through, please do that as soon as you can so we can move ahead without any errors.

**Review by madeofhuman — APPROVED** [review]

**Review by tonymontaro — APPROVED** [review]
Good job.

**Pull request merged**